components/rhs_card_header/rhs_card_header.tsx

```diff
@@ -10,21 +9,21 @@ import OverlayTrigger from 'components/overlay_trigger';
 
 import Constants, {RHSStates} from 'utils/constants';
 
-export default class RhsCardHeader extends React.PureComponent {
-    static propTypes = {
-        previousRhsState: PropTypes.oneOf(Object.values(RHSStates)),
-        isExpanded: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            showMentions: PropTypes.func,
-            showSearchResults: PropTypes.func,
-            showFlaggedPosts: PropTypes.func,
-            showPinnedPosts: PropTypes.func,
-            closeRightHandSide: PropTypes.func,
-            toggleRhsExpanded: PropTypes.func.isRequired,
-        }),
+type Props = {
+    previousRhsState: string;
+    isExpanded: boolean;
+    actions: {
+        showMentions: () => void;
+        showSearchResults: () => void;
+        showFlaggedPosts: () => void;
+        showPinnedPosts: () => void;
+        closeRightHandSide: () => void;
+        toggleRhsExpanded: () => void;
     };
+};
 
-    handleBack = (e) => {
+export default class RhsCardHeader extends React.PureComponent<Props> {
+    handleBack = (e: { preventDefault: () => void }): void => {
```

Comment:
I feel like there is a better way to type `e`. What would be the preferred type?

Reply:
Should be of type `React.MouseEvent<HTMLAnchorElement>`